File: internal/content/testdata/bench/typescript/matching-colored-houses.ts
function match_colored_houses(left_colors: string[], right_colors: string[]): number[] {
    const color_to_pos: {[key: string]: number} = {};
    for (let i = 0; i < right_colors.length; i++) {
        color_to_pos[right_colors[i]] = i + 1;
    }
    return left_colors.map(c => color_to_pos[c]);
}

File: internal/content/testdata/bench/go/matching-colored-houses.go
func match_colored_houses(left_colors []int, right_colors []int) []int {
	colorToPos := make(map[int]int)
	for i, c := range right_colors {
		colorToPos[c] = i + 1
	}
	result := make([]int, len(left_colors))
	for i, c := range left_colors {
		result[i] = colorToPos[c]
	}
	return result
}
